projects/solar/test/test_data_store.c

```diff
@@ -6,47 +6,85 @@
+  TEST_ASSERT_OK(data_store_get(VALID_TEST_DATA_POINT, &value));
+  TEST_ASSERT_EQUAL(VALID_TEST_VALUE, value);
+}
+
+// Test that we can test whether a value is set correctly.
+void test_data_store_get_is_set_basic(void) {
+  TEST_ASSERT_OK(data_store_init());
+  bool is_set;
+  TEST_ASSERT_OK(data_store_get_is_set(VALID_TEST_DATA_POINT, &is_set));
+  TEST_ASSERT_EQUAL(false, is_set);
+  TEST_ASSERT_OK(data_store_set(VALID_TEST_DATA_POINT, VALID_TEST_VALUE));
+  TEST_ASSERT_OK(data_store_get_is_set(VALID_TEST_DATA_POINT, &is_set));
+  TEST_ASSERT_EQUAL(true, is_set);
 }
 
 // Test that every data point works.
-void test_data_store_set_and_get_thorough(void) {
+void test_data_store_set_and_get_and_get_is_set_thorough(void) {
+  TEST_ASSERT_OK(data_store_init());
   uint16_t value;
+  bool is_set;
   for (DataPoint data_point = 0; data_point < NUM_DATA_POINTS; data_point++) {
-    TEST_ASSERT_OK(data_store_set(data_point, 0xDEAD));
+    TEST_ASSERT_OK(data_store_get_is_set(data_point, &is_set));
+    TEST_ASSERT_EQUAL(false, is_set);
+    TEST_ASSERT_OK(data_store_set(data_point, VALID_TEST_VALUE));
+    TEST_ASSERT_OK(data_store_get_is_set(data_point, &is_set));
+    TEST_ASSERT_EQUAL(true, is_set);
     TEST_ASSERT_OK(data_store_get(data_point, &value));
-    TEST_ASSERT_EQUAL(0xDEAD, value);
+    TEST_ASSERT_EQUAL(VALID_TEST_VALUE, value);
   }
 }
 
 // Test that |data_store_set| responds correctly on an invalid index.
 void test_data_store_set_invalid_index(void) {
+  TEST_ASSERT_OK(data_store_init());
   TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, data_store_set(NUM_DATA_POINTS, 0));
 }
 
 // Test that |data_store_get| responds correctly on an invalid index.
 void test_data_store_get_invalid_index(void) {
+  TEST_ASSERT_OK(data_store_init());
   uint16_t value;
   TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, data_store_get(NUM_DATA_POINTS, &value));
 }
 
 // Test that |data_store_get| responds correctly on a null pointer.
 void test_data_store_get_null_pointer(void) {
-  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, data_store_get(DATA_POINT_VOLTAGE_1, NULL));
+  TEST_ASSERT_OK(data_store_init());
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, data_store_get(VALID_TEST_DATA_POINT, NULL));
+}
+
+// Test that |data_store_get_is_set| responds correctly on an invalid index.
+void test_data_store_get_is_set_invalid_index(void) {
+  TEST_ASSERT_OK(data_store_init());
+  bool is_set;
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, data_store_get_is_set(NUM_DATA_POINTS, &is_set));
+}
+
+// Test that |data_store_get_is_set| responds correctly on a null pointer.
+void test_data_store_get_is_set_null_pointer(void) {
+  TEST_ASSERT_OK(data_store_init());
+  TEST_ASSERT_EQUAL(STATUS_CODE_INVALID_ARGS, data_store_get_is_set(VALID_TEST_DATA_POINT, NULL));
 }
 
 // Test that |data_store_done| correctly raises DATA_READY_EVENTs.
 void test_data_store_done_raises_events(void) {
+  TEST_ASSERT_OK(data_store_init());
```

Comment:
nit: technically you don't need this here, do you?

Reply:
Technically no, but it's an implementation detail of the module that `data_store_done` doesn't depend on `data_store_init`, so I prefer not to depend on that.